modal/mount.py

```diff
@@ -324,25 +324,39 @@ def add_local_dir(
         condition: Optional[Callable[[str], bool]] = None,
         # add files from subdirectories as well
         recursive: bool = True,
+        include_files: Optional[Callable[[Path], bool]] = None,
     ) -> "_Mount":
         """
         Add a local directory to the `Mount` object.
         """
+
+        if condition is not None:
+            deprecation_warning((2024, 12, 10), "`condition` is deprecated, please use `include_files` instead")
```

Comment:
Since we plan to deprecate `Mount` as public-facing API I'm not sure it's worth the churn to substitute the new name here.

Reply:
Yeah makes sense, think we should just let condition be as is & set a deprecation warning on mounts as a whole later instead?